### PR TITLE
Feature: Upload external datasets

### PR DESCRIPTION
**Pull request by duranb**
Partially resolves https://github.com/NASA-AMMOS/aerie-ui/issues/1424

Refer to [UI PR](https://github.com/NASA-AMMOS/aerie-ui/pull/1492) for testing procedure.

### CODE REVIEW

File: src/packages/plan/plan.ts [hunk 1]
```diff
@@ -339,7 +332,293 @@ export async function importPlan(req: Request, res: Response) {
...
+            fileStream.pipe(parser);
+          });
+
+          // Keep track of the time column's index separately since the name of the column is static
+          let timeColumnIndex = -1;
+
+          // Create a lookup for the profile name's index in each CSV row
+          const headerIndexMap: Record<string, number> = parsedCSV[0].reduce(
+            (prevHeaderIndexMap: Record<string, number>, header: string, headerIndex: number) => {
+              if (new RegExp(timeColumnKey).test(header)) {
```
Comment (dandelany) [inline src/packages/plan/plan.ts:421]: One thing to note here is that this will loosely match any header that contains the **substring** `time_utc`, eg. `event_time_utc` will match. I think this is **OK** here since there shouldn't be any ambiguity, since there should only be one timestamp column & the others should be numeric, but worth calling out & considering if it should be a strict equality match

File: src/util/time.test.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  });
+
+  test('getTimeDifference', () => {
+    expect(getTimeDifference('2024-01-01T00:10:00', '2024-01-01T00:11:00', 6)).toEqual(60000000);
+    expect(getTimeDifference('2024-01-01T00:01:00', '2024-01-01T00:11:00', 6)).toEqual(600000000);
+    expect(getTimeDifference('2024-245T00:01:00.0', '2024-245T00:02:00.0', 6)).toEqual(60000000);
+    expect(getTimeDifference('2024-245T00:01:00.0', '2024-245T12:02:00.0', 6)).toEqual(43260000000);
+    expect(getTimeDifference('2024-243T00:01:00.0', '2024-245T12:02:00.0', 6)).toEqual(216060000000);
+  });
+});
```
Comment (dandelany) [inline src/util/time.test.ts:55]: Could use some more unit tests here for better coverage of the parsing functions in `time.ts`

**Review by dandelany — APPROVED** [review]
This looks good - a few non-blocking notes below, but I'm going to approve and merge this for today's release. I've tested locally with the examples and all seems to be working correctly. We still need to document this feature and potentially add some better tests & coverage for some edge cases/handling for other data types, but this is a first pass at an experimental feature that users are asking for, so I think it's worth releasing v1 as is. Thanks @duranb !
